Packet++/src/HttpLayer.cpp

```diff
@@ -209,7 +209,7 @@ HttpRequestFirstLine::HttpRequestFirstLine(HttpRequestLayer* httpRequest) : m_Ht
 }
 
 HttpRequestFirstLine::HttpRequestFirstLine(HttpRequestLayer* httpRequest, HttpRequestLayer::HttpMethod method, HttpVersion version, std::string uri)
-		throw(HttpRequestFirstLineException)
+try		// throw(HttpRequestFirstLineException)

```

Comment:
Isn't it better to wrap the try-catch clause with curly brackets (in the beginning and end of the constructor?)

Reply:
@seladb Please take a look on this video https://youtu.be/vmtGCJbNKNE  
I personally think constructors of type:  
```
ClassName(args)
try 
{ /*...smth...*/}
catch(const KnownException& e)
{ /*...smth with exception and resources...*/ }
...
catch(...) 
{/*...smth with unknown exception..*/}
```
and
```
ClassName(args) 
{
    try 
    { /*...smth...*/}
    catch(const KnownException& e)
    { /*...smth with exception and resources...*/ }
    ...
    catch(...) 
    {/*...smth with unknown exception..*/}
}
```  
are not totally same from position of c++ run-time library and compiler. But i may be wrong.  
If You find a reason why any of this constructions are better please leave a response with links here.

Reply:
I actually didn't know about function-try-blocks, thanks for the free c++ lesson :)
After reading about it a little and watching your video I think that function-try-blocks in c'tors are best for catching exceptions thrown by either by initializers or base-class c'tor. Both are not the case here. Also, function-try-blocks can't really catch the exception (as they have to end with `throw`) but they allow you to do some intermediate handling before the exception is thrown. In this case there is no intermediate handling in the catch clause - we just throw or terminate.
Having said all of that - there's nothing wrong in keeping that code, it may have some value in the future.
I leave it for you to decide. Either way I'll approve your PR. Please let me know what you think

Reply:
@seladb Ok, I am happy about that You can learn something new from this PR. By now I have no intention to change something so if You don't mind please approve merge.